database.c

```diff
@@ -60,6 +62,12 @@ static double get_db_filesize(void)
 	return 1e-6*st.st_size;
 }
 
+static bool file_exists(const char *filename)
+{
+	struct stat st;
+	return (stat(filename, &st) != 0);
```

Comment:
`stat` returns zero if the call is successful (the file exists). This should then check for equality with zero instead of inequality.

Also, the parentheses are unnecessary